client/my-sites/plans-grid/components/features-grid/index.tsx

```diff
@@ -211,12 +213,24 @@ class FeaturesGrid extends Component< FeaturesGridType > {
 			} );
 	}
 
-	renderMobileFreeDomain( planSlug: PlanSlug, isMonthlyPlan?: boolean ) {
+	renderMobileFreeDomain( gridPlan: GridPlan ) {
 		const { translate } = this.props;
+		const { planSlug, isMonthlyPlan } = gridPlan;
 
 		if ( isMonthlyPlan || isWpComFreePlan( planSlug ) || isWpcomEnterpriseGridPlan( planSlug ) ) {
 			return null;
 		}
+
+		// Remove the custom domain feature for Woo Express plans with introductory offer.
+		if (
+			isWooExpressPlan( planSlug ) &&
+			! gridPlan.features.wpcomFeatures.some(
```

Comment:
We're leveraging the introductory offer check from [here](https://github.com/Automattic/wp-calypso/blob/496d908885f27adae0fbf9376debe375338a3f1b/client/my-sites/plans-grid/hooks/npm-ready/data-store/use-plan-features-for-grid-plans.ts#L100-L109)